### PR TITLE
Update Dependencies, Fixes #1

### LINKED ISSUE #1: Pre- and Post-Processing Servers Shouldn't Be Required
It should be possible to use Remedi4J without the pre- and post-processing servers (e.g. if you're doing the processing locally and just want to do the translation via REMEDI), but currently the client throughs a NullPointerException if you try to do this.

### PR DESCRIPTION
**Pull request by jamesdbaker**
Fixes issue #1 and updates dependencies to latest versions (including security fixes)

### CODE REVIEW

File: src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java [hunk 1]
```diff
@@ -38,43 +38,58 @@ National Crime Agency (c) Crown Copyright 2018
...
    * @param translationServer
-   *    URI of the translation server or load balancer
+   *    URI of the translation server or load balancer (must not be null)
    * @param postProcessingServer
-   *    URI of the post-processing server
+   *    URI of the post-processing server (can be null)
    */
   public RemediClient(URI preProcessingServer, URI translationServer, URI postProcessingServer){
-    wsPreProcessingServer = HttpClient.newHttpClient()
+    if(preProcessingServer != null)
```
Comment (richc-nca) [inline src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java]: if(preProcessingServer != null) {
}

Style consistency in the file

File: src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java [hunk 2]
```diff
@@ -147,6 +179,9 @@ public RemediClient(URI preProcessingServer, URI translationServer, URI postProc
    *    The response from the pre-processor
    */
   public CompletableFuture<PreProcessorResponse> preProcess(String language, String text) {
+    if(wsPreProcessingServer == null)
```
Comment (richc-nca) [inline src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java]: As above brackets for style consistency?

File: src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java [hunk 3]
```diff
@@ -217,6 +252,9 @@ public RemediClient(URI preProcessingServer, URI translationServer, URI postProc
    *    The response from the post-processor
    */
   public CompletableFuture<PostProcessorResponse> postProcess(String language, String text) {
+    if(wsPostProcessingServer == null)
```
Comment (richc-nca) [inline src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java]: As above for style consistency

File: src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java [hunk 4]
```diff
@@ -267,7 +305,17 @@ public RemediClient(URI preProcessingServer, URI translationServer, URI postProc
 
   @Override
   public void close() {
+    if(wsPreProcessingServer != null){
+      wsPreProcessingServer.sendClose(1000, "Client closed");
```
Comment (richc-nca) [inline src/main/java/uk/gov/nca/remedi4j/client/RemediClient.java]: Use WebSocket.NORMAL_CLOSURE instead of 1000?

**Review by richc-nca — APPROVED** [review]

**Review by richc-nca — APPROVED** [review]

**Comment by NCA-tbyatt** [comment]
_

**Pull request merged**